lib/versions.ps1

```diff
@@ -1,53 +1,258 @@
 # versions
-function latest_version($app, $bucket, $url) {
-    (manifest $app $bucket $url).version
-}
-function current_version($app, $global) {
-    @(versions $app $global)[-1]
+function Get-LatestVersion {
+    <#
+    .SYNOPSIS
+        Get latest version of app from manifest
+    .PARAMETER AppName
+        App's name
+    .PARAMETER Bucket
+        Bucket which the app belongs to
+    .PARAMETER Uri
+        Remote app manifest's URI
+    #>
+    [OutputType([String])]
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory = $true, Position = 0, ValueFromPipeline = $true)]
+        [Alias('App')]
+        [String]
+        $AppName,
+        [Parameter(Position = 1)]
+        [String]
+        $Bucket,
+        [Parameter(Position = 2)]
+        [String]
+        $Uri
+    )
+    return (manifest $AppName $Bucket $Uri).version
 }
-function versions($app, $global) {
-    $appdir = appdir $app $global
-    if(!(test-path $appdir)) { return @() }
 
-    sort_versions (Get-ChildItem $appdir -dir -attr !reparsePoint | Where-Object { $null -ne $(Get-ChildItem $_.fullname) } | ForEach-Object { $_.name })
+function Select-CurrentVersion {
+    <#
+    .SYNOPSIS
+        Select current version of installed app, from 'current\manifest.json' or modified time of version directory
+    .PARAMETER AppName
+        App's name
+    .PARAMETER Global
+        Globally installed application
+    #>
+    [OutputType([String])]
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory = $true, Position = 0, ValueFromPipeline = $true)]
+        [Alias('App')]
+        [String]
+        $AppName,
+        [Parameter(Position = 1)]
+        [Switch]
+        $Global
+    )
+
+    $appPath = appdir $AppName $Global
+    if (Test-Path "$appPath\current") {
+        $currentVersion = (installed_manifest $AppName 'current' $Global).version
+    } else {
+        $installedVersion = Get-InstalledVersion -AppName $AppName -Global:$Global
+        if ($installedVersion) {
```

Comment:
This will be always true

Reply:
<img width="327" alt="if (@())" src="https://user-images.githubusercontent.com/5832170/71363360-d6c84e00-25d3-11ea-8b26-6d719356f7b6.png">